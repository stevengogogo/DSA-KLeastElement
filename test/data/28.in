10 50000
-1236 -5835 5929 -699 8031 -6404 8992 -2198 -4601 260
Query 8 9 2
Query 1 4 1
Query 5 9 5
Delete 10
Reverse 5 6
Query 2 5 2
Insert 4 -6143
Reverse 10 10
Reverse 6 8
Insert 11 6513
Insert 10 2242
Query 6 12 1
Insert 9 -1526
Reverse 3 7
Reverse 2 8
Reverse 13 13
Insert 9 -3029
Reverse 8 11
Insert 4 5830
Insert 15 3975
Reverse 12 14
Delete 10
Delete 7
Query 3 7 5
Insert 10 -645
Reverse 13 15
Delete 11
Delete 10
Query 13 13 1
Insert 3 -5157
Query 6 7 1
Reverse 9 11
Insert 9 9314
Delete 15
Reverse 2 14
Delete 9
Reverse 12 13
Reverse 3 11
Query 1 13 12
Query 6 6 1
Insert 4 7455
Reverse 10 14
Delete 11
Insert 7 -5429
Insert 10 7616
Query 12 14 3
Query 7 14 1
Query 15 15 1
Query 14 15 1
Query 1 6 6
Insert 6 -3098
Query 16 16 1
Delete 12
Insert 16 8559
Insert 2 9759
Insert 12 -4820
Delete 11
Insert 15 -9485
Insert 16 -2418
Delete 13
Insert 16 5668
Delete 8
Reverse 13 17
Insert 18 -3195
Query 17 18 1
Delete 13
Reverse 13 14
Insert 7 6279
Query 18 18 1
Query 3 13 1
Reverse 11 18
Reverse 14 17
Delete 18
Insert 6 4023
Delete 15
Reverse 12 16
Query 15 17 3
Delete 10
Delete 5
Delete 16
Reverse 6 7
Delete 7
Delete 9
Reverse 13 13
Delete 12
Insert 12 6140
Delete 8
Insert 4 -1399
Query 1 1 1
Delete 4
Insert 13 7697
Reverse 13 13
Reverse 3 10
Insert 12 -5514
Reverse 12 13
Reverse 3 6
Insert 2 -4854
Reverse 5 6
Delete 5
Delete 1
Reverse 5 11
Delete 2
Query 3 10 7
Delete 3
Insert 4 8794
Reverse 5 5
Reverse 8 10
Insert 3 8547
Delete 9
Query 8 9 1
Query 2 9 5
Reverse 8 9
Delete 7
Delete 8
Delete 8
Insert 9 6078
Delete 8
Reverse 5 9
Delete 6
Query 7 8 1
Reverse 4 5
Query 3 4 2
Reverse 7 8
Reverse 6 8
Delete 7
Query 5 6 2
Query 7 7 1
Reverse 2 7
Delete 7
Reverse 6 6
Query 2 5 2
Query 3 3 1
Delete 1
Reverse 2 5
Reverse 2 3
Reverse 1 4
Reverse 3 5
Reverse 5 5
Query 3 4 2
Query 2 3 1
Query 4 4 1
Insert 1 -455
Insert 6 5812
Reverse 5 7
Insert 4 5734
Insert 7 -4369
Query 4 6 1
Reverse 9 9
Delete 1
Query 3 4 1
Insert 2 -9820
Delete 9
Delete 7
Reverse 2 7
Insert 2 -341
Query 3 8 5
Reverse 4 5
Delete 6
Query 2 5 4
Insert 4 1310
Reverse 3 6
Query 5 6 2
Query 1 8 6
Insert 2 -564
Delete 9
Insert 2 2169
Reverse 4 7
Delete 2
Insert 3 -6991
Delete 4
Delete 4
Query 3 7 3
Query 4 7 1
Delete 6
Insert 7 4553
Insert 4 -6249
Query 3 5 1
Reverse 8 8
Insert 2 5003
Reverse 8 8
Reverse 1 8
Insert 4 8120
Delete 9
Insert 9 941
Delete 4
Delete 2
Query 3 7 2
Insert 5 642
Insert 4 -362
Query 4 5 1
Query 7 8 1
Reverse 3 9
Delete 3
Reverse 5 7
Query 3 9 3
Reverse 8 8
Delete 9
Delete 4
Query 5 7 3
Reverse 6 6
Query 4 6 2
Delete 4
Delete 6
Query 2 5 1
Reverse 4 5
Reverse 5 5
Query 4 5 1
Query 2 2 1
Insert 5 262
Delete 4
Delete 3
Insert 2 -1838
Delete 4
Reverse 4 4
Delete 2
Insert 2 -9634
Delete 2
Reverse 1 1
Insert 4 5653
Query 1 1 1
Insert 5 3897
Insert 3 6694
Delete 3
Delete 1
Query 3 4 1
Reverse 4 4
Reverse 4 4
Insert 2 -9339
Delete 1
Insert 5 -52
Insert 2 -3778
Insert 4 -3799
Reverse 1 2
Insert 4 1886
Query 1 6 5
Delete 4
Insert 8 7031
Query 1 3 1
Query 6 6 1
Reverse 7 7
Query 5 6 1
Insert 3 9397
Insert 6 6172
Delete 6
Delete 1
Delete 8
Delete 7
Reverse 6 6
Reverse 4 5
Insert 2 7049
Reverse 2 4
Insert 3 -181
Insert 5 8990
Reverse 6 8
Delete 6
Query 5 5 1
Insert 1 -7441
Query 6 8 2
Insert 7 -9945
Insert 2 -8592
Reverse 9 9
Insert 4 -5792
Reverse 7 11
Delete 12
Delete 5
Query 8 8 1
Insert 2 9545
Reverse 8 9
Query 2 2 1
Delete 3
Reverse 5 5
Insert 5 3246
Query 5 9 2
Query 7 10 4
Query 10 11 1
Delete 10
Reverse 2 4
Delete 2
Query 8 9 2
Delete 7
Delete 3
Delete 6
Reverse 3 5
Query 1 3 2
Reverse 3 4
Query 2 3 2
Reverse 3 3
Insert 1 6437
Delete 2
Insert 2 -229
Insert 3 930
Query 5 7 1
Insert 7 9442
Delete 2
Delete 4
Reverse 1 5
Insert 6 5434
Delete 5
Insert 2 -2448
Query 2 5 1
Insert 4 8452
Delete 8
Reverse 8 8
Reverse 1 2
Insert 5 6563
Delete 6
Reverse 5 6
Reverse 1 6
Delete 6
Query 1 5 5
Query 3 4 2
Query 6 6 1
Query 2 3 2
Reverse 5 7
Delete 3
Reverse 1 5
Reverse 5 6
Query 5 5 1
Delete 4
Reverse 2 5
Delete 2
Insert 5 253
Query 3 3 1
Reverse 2 3
Delete 5
Insert 5 4567
Reverse 4 4
Query 3 3 1
Query 2 3 1
Reverse 4 4
Insert 2 -3710
Delete 6
Insert 5 -1617
Query 3 4 2
Reverse 3 4
Reverse 5 6
Delete 4
Query 3 4 2
Delete 4
Delete 1
Query 3 3 1
Delete 2
Delete 1
Query 1 1 1
Delete 1
