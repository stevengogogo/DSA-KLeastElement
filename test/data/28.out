-2198
-5835
8992
-5835
-6404
8031
-5835
-6143
8031
8031
6513
-5429
-3029
-3029
7455
-3029
-9485
-3195
-5429
-2198
-1236
6279
5929
6140
6140
8547
6140
6279
6140
6140
6279
6140
6279
5734
5734
7697
7697
5812
6140
5812
-341
-6991
-6249
-6991
-564
-564
642
-362
-6991
-6991
7697
8547
3897
1886
-9339
3897
3897
8990
7049
-9945
9545
-181
8990
8990
9397
-7441
-181
-181
-9339
9442
8452
5434
8452
9442
7049
9442
7049
9442
9442
-1617
-1617
